### PR TITLE
feat: ensure global hidden attribute is honored for all components

### PR DESCRIPTION
**Pull request by Elijbet**
**Related Issue:** #3821

## Summary

Added support for the global `hidden` attribute for all components for behavior consistency.

Made sure all components use `hidden` test util.

### CODE REVIEW

File: src/components/loader/loader.scss [hunk 1]
```diff
@@ -47,11 +47,17 @@ $loader-circumference: ($loader-scale - (2 * $stroke-width)) * 3.14159;
 }
 
 :host {
-  @apply hidden;
-}
+  visibility: hidden;
 
-:host([active]) {
-  @apply flex;
+  &:host([hidden]) {
```
Comment (jcfranco) [inline src/components/loader/loader.scss:52]: Please revert these changes. Remember we discussed a separate PR to ensure the `hidden` attribute gets applied to all components (in `includes.scss`).

**Review by driskull — APPROVED** [review]
👀 Looks good to me 👍

File: src/components/accordion/accordion.e2e.ts [hunk 2]
```diff
@@ -15,6 +15,8 @@ describe("calcite-accordion", () => {
   `;
   it("renders", async () => renders("calcite-accordion", { display: "block" }));
 
+  it("honors hidden attribute", async () => hidden("calcite-accordion"));
```
Comment (jcfranco) [inline src/components/accordion/accordion.e2e.ts:18]: Components that are hidden by default (using the `calcite-hydrated-hidden` attribute) should use initially-visible HTML for testing. Most if not all of our test utils take a tag or HTML (see [`disabled` example](https://github.com/Esri/calcite-components/blob/33cbffe3d213ed0d9b362b8631a3d22cc5bda846/src/components/block/block.e2e.ts#L46-L47)).

File: src/components/input-date-picker/input-date-picker.e2e.ts [hunk 3]
```diff
@@ -8,6 +16,8 @@ const animationDurationInMs = 200;
 describe("calcite-input-date-picker", () => {
   it("renders", async () => renders("calcite-input-date-picker", { display: "inline-block" }));
 
+  it("honors hidden attribute", async () => hidden("calcite-date-picker"));
```
Comment (jcfranco) [inline src/components/input-date-picker/input-date-picker.e2e.ts:19]: This should be `calcite-input-date-picker`.

File: src/components/radio-group-item/radio-group-item.e2e.ts [hunk 4]
```diff
@@ -1,9 +1,11 @@
 import { newE2EPage } from "@stencil/core/testing";
-import { renders } from "../../tests/commonTests";
+import { renders, hidden } from "../../tests/commonTests";
 
 describe("calcite-radio-group-item", () => {
   it("renders", () => renders("calcite-radio-group-item", { display: "flex" }));
 
+  it("honors hidden attribute", async () => hidden("calcite-link"));
```
Comment (jcfranco) [inline src/components/radio-group-item/radio-group-item.e2e.ts:7]: This should be `calcite-radio-group-item`.

File: src/components/tooltip/tooltip.e2e.ts [hunk 5]
```diff
@@ -11,6 +11,8 @@ describe("calcite-tooltip", () => {
     });
   });
 
+  it("honors hidden attribute", async () => hidden("calcite-tooltip"));
```
Comment (jcfranco) [inline src/components/tooltip/tooltip.e2e.ts:14]: Remove. This is a duplicate test (missing `import` in the diff was the giveaway).

**Review by jcfranco — APPROVED** [review]
Aside from some comments, this LGTM! 🎉

File: src/components/accordion/accordion.e2e.ts (hunk 2 above)
Comment (Elijbet) [reply src/components/accordion/accordion.e2e.ts:18]: I think you may have meant this comment for calcite-accordion-item.

Comment (jcfranco) [reply src/components/accordion/accordion.e2e.ts:18]: Apologies. I thought I commented on `calcite-alert`.
